docs/markdown/Reference-manual.md

```diff
@@ -2714,6 +2716,15 @@ library. This object has the following methods:
   object will only inherit other attributes from its parent as
   controlled by keyword arguments.
 
+### File object
+
+*(since 0.58.0)*
+
+This object is an entry in the file array returned by [`files()`](#files). This
+object contains the following methods:
+
+- `full_path()`: returns a string of the full path to the file
```

Comment:
This should have a warning of something like: "Use this method with caution. The return value is a string and carries no dependency information. Thus if you use it as an argument to a command, Meson does not know that it refers to an actual file and can not set up dependencies properly. Whenever possible you should use the file object directly in the command line. It will be autoexpanded to point to the file and all necessary dependencies are set up automatically."